cmake/AwsCryptosdkCFlags.cmake

```diff
@@ -74,6 +75,7 @@ function(aws_cryptosdk_set_common_properties target)
         list(APPEND AWS_C_FLAGS -Wno-long-long)
         list(APPEND AWS_C_FLAGS -Wno-missing-field-initializers)
         list(APPEND AWS_C_FLAGS -Wno-shorten-64-to-32)
+        list(APPEND AWS_C_FLAGS -Wno-deprecated-declarations)
```

Comment:
Question: Is there anyway to make this conditional on OpenSSL-3.x?
Is there anyway to Scope this ignore to OpenSSL?

Reply:
As far as I can tell, no.